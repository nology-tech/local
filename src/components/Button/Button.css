@import url("https://fonts.googleapis.com/css2?family=Inter&family=Lato&family=Roboto&display=swap");
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

/*                          Colours                         */
/*                          Font stuff              */
* {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande", "Lucida Sans", Arial, sans-serif;
  color: #000000;
}

h1,
h2,
h3,
h4,
h5,
h6,
p {
  margin: 0.5rem 0;
  display: inline-block;
  width: 100%;
}

button {
  font-size: 1rem;
}

h1 {
  font-size: 2.1rem;
}

h2 {
  font-size: 1.65rem;
}

h3 {
  font-size: 1.25rem;
}

@media screen and (min-width: 768px) {
  :root {
    font-size: 110%;
  }
}
@media screen and (min-width: 1024px) {
  :root {
    font-size: 120%;
  }
}
.button {
  font-family: "Inter";
  background-color: #1f1b1b;
  width: 9rem;
  height: 3rem;
  font-size: 1rem;
  border: 0.1rem solid black;
  opacity: 0.8;
}
.button:hover {
  opacity: 1;
  transition: 0.3s;
}
.button:active {
  border: 0.2rem solid #f3f2e8;
  opacity: 0.4;
}

.button--contact {
  height: 4rem;
  margin: auto;
  width: 12rem;
  color: #f3f2e8;
  background-color: #5bb27c;
  border-color: #5bb27c;
  font-size: 3rem;
  font-weight: 400;
  border-radius: 1.7rem;
}/*# sourceMappingURL=Button.css.map */