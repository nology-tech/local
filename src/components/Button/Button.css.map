{"version":3,"sources":["../../styles/_typography.scss","../../styles/_reset.scss","Button.css","../../styles/_variables.scss","Button.scss"],"names":[],"mappings":"AACQ,oGAAA;ACDR;EACE,UAAA;EACA,SAAA;EACA,sBAAA;ACEF;;ACEA,6DAAA;AAuBA,qDAAA;AHxBA;EACE,qGAAA;EAEA,cGCW;ADEb;;AFAA;;;;;;;EAQE,gBAAA;EACA,qBAAA;EACA,WAAA;AEEF;;AFCA;EACE,eAAA;AEEF;;AFCA;EACE,iBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EAEE;IACE,eAAA;EECF;AACF;AFEA;EAEE;IACE,eAAA;EEDF;AACF;AE9CA;EACE,oBD6BiB;EC5BjB,yBDIiB;ECFjB,WAAA;EAEA,YAAA;EAEA,eD+BkB;EC7BlB,0BAAA;EACA,YAAA;AF4CF;AE3CE;EACE,UAAA;EACA,gBAAA;AF6CJ;AE3CE;EACE,4BAAA;EACA,YAAA;AF6CJ;;AEzCA;EACE,YAAA;EACA,YAAA;EACA,YAAA;EACA,cDVM;ECWN,yBDRoB;ECSpB,qBDToB;ECUpB,eAAA;EAEA,gBDIoB;ECFpB,qBAAA;AF0CF","file":"Button.css"}