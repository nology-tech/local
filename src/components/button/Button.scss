@use "../../../src/styles/reset" as *;
@use "../../../src/styles/variables" as *;
@use "../../styles/typography" as *;
@use "../../styles/mixins" as *;

.button {
  background-color: $background-color;

  width: 9rem;

  height: 3rem;

  font-size: $default-text-size;

  padding: 10px 20px;

  border: solid black;
}

.button-contact {
  margin: auto;
  width: 12rem;
  font-family: $text-Font-Family;
  color: $white;
  margin-top: 10px;
  padding: 0.5rem 0.9rem;

  background-color: $green-color-palette;

  font-size: 3rem;

  font-weight: $default-font-weight;

  border-radius: 25px;

  border: none;
}
