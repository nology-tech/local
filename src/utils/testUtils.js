import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
