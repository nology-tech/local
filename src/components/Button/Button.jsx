import "./Button.scss";

const Button = ({ buttonText, buttonName, onClick }) => {
  let buttonClass = `button button--${buttonName}`;

  return (
    <button className={buttonClass} onClick={onClick}>
      {buttonText}
    </button>
  );
};

export default Button;
