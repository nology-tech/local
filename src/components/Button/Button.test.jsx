import Button from "./Button";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

it("should render the button page", () => {
  const { container } = customRender(<Button />);

  expect(container).toMatchSnapshot();
});

it("should call the functions within onclick when user clicks", () => {
  const mockFunction = jest.fn();
  customRender(<Button onClick={mockFunction} />);

  const button = screen.getByRole("button");

  userEvent.click(button);

  expect(mockFunction).toBeCalled();
});
