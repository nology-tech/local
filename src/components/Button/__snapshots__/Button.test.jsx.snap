// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the button page 1`] = `
<div>
  <button
    class="button button--undefined"
  >
    <img
      class="button--image--undefined"
    />
  </button>
</div>
`;
